not assignable to parameter of type Dependency

ef27de9 reproduces this compilation problem that I had downstream with code that compiled a few months ago, that apparently wasn't properly package-locked

```
    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/Dependency.spec.ts:123:16 - error TS2345: Argument of type 'DependencyPromise<{}>' is not assignable to parameter of type 'Dependency<string>'.
      Types of property 'then' are incompatible.
        Type '<TResult1 = {}, TResult2 = never>(onfulfilled?: (value: {}) => TResult1 | PromiseLike<TResult1>, ...' is not assignable to type '<TResult1 = string, TResult2 = never>(onfulfilled?: (value: string) => TResult1 | PromiseLike<TRe...'.
          Types of parameters 'onfulfilled' and 'onfulfilled' are incompatible.
            Types of parameters 'value' and 'value' are incompatible.
              Type '{}' is not assignable to type 'string'.

    123       receiver(p);
                       ~
```

--- a/src/Dependency.spec.ts
+++ b/src/Dependency.spec.ts
@@ -108,4 +108,22 @@ describe("DependencyInjectionLazy", () => {
 
   });
 
+  describe("Instantiation as-is", () => {
+
+    let received = null;
+
+    let receiver = function(depPropmise: Dependency<string>) {
+      received = depPropmise;
+    }
+
+    it("Type declarations are ok with a promise from new without generic type", () => {
+      let p = new DependencyPromise((resolve, reject) => {
+        resolve("Resolved!");
+      });
+      receiver(p);
+      expect(received).toBe(p);
+    });
+
+  });
+
 });

--- a/src/Dependency.spec.ts
+++ b/src/Dependency.spec.ts
@@ -117,7 +117,7 @@ describe("DependencyInjectionLazy", () => {
     }
 
     it("Type declarations are ok with a promise from new without generic type", () => {
-      let p = new DependencyPromise((resolve, reject) => {
+      let p = new DependencyPromise<string>((resolve, reject) => {
         resolve("Resolved!");
       });
       receiver(p);

